File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> mpp=new Dictionary<int,int>();
        for(int i=0;i<nums.Length;i++){
            int a=nums[i];
            int more=target-nums[i];
            if(mpp.ContainsKey(more)){
                return new int[] {mpp[more],i};
            }
            mpp[a]=i;
        }
        return new int[] {0};
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>ans;

        unordered_map<int,int>mpp;
        for(int i=0;i<nums.size();i++){
            int rem=target-nums[i];

            if(mpp.find(rem)!=mpp.end()){
                ans.push_back(mpp.find(rem)->second);
                ans.push_back(i);
                break;
            }

            mpp.insert(pair<int,int>(nums[i],i));
        }
        return ans;
    }
};